pkg/client/client.go

```diff
@@ -1068,6 +1069,55 @@ func (c *KpmClient) ParseOciOptionFromString(oci string, tag string) (*opt.OciOp
 	return ociOpt, nil
 }
 
+// PrintDependencyGraph will print the dependency graph of kcl package dependencies
+func (c *KpmClient) PrintDependencyGraph(kclPkg *pkg.KclPkg) error {
+	_, depGraph, err := c.downloadDeps(kclPkg.ModFile.Dependencies, kclPkg.Dependencies)
+	if err != nil {
+		return err
+	}
+
+	sources, err := FindSource(depGraph)
+	if err != nil {
+		return err
+	}
+
+	// add the root vertex(package name) to the dependency graph.
+	root := fmt.Sprintf("%s@%s", kclPkg.GetPkgName(), kclPkg.GetPkgVersion()) 
```

Comment:
This graph should be printed to the kpm client logger.

Reply:
the graph is getting printed on the cli side https://github.com/kcl-lang/kpm/blob/6fcfb5c0b21f92a79c744bab39d78fd08b646691/pkg/cmd/cmd_graph.go#L77

if you want me to transfer the print logic to client side itself, please let me know.